### DISCUSSION TITLE
RootPainter.exe not responding

### DISCUSSION THREAD

**Discussion by qielax** [Abe404/root_painter#127] [Q&A]
Hi, Abraham and community
I encontered some problems when using rootpainter client. After I clicked "Save & Next >", the button stayed at "loading.."(for a long time). Then, if I click the window, it will show "not responding". (in these screenshots bellow, "未响应" means "not responding") 
<img width="956" alt="p1" src="https://github.com/Abe404/root_painter/assets/150230377/b0a5bf89-8972-47eb-8c21-54873cbc7b6b">
<img width="949" alt="p2" src="https://github.com/Abe404/root_painter/assets/150230377/4df7e7e1-63f2-4bc6-a5f0-acb056030d24">
<img width="951" alt="p3" src="https://github.com/Abe404/root_painter/assets/150230377/4a43bf7d-b811-4e97-9cab-23e75a45affd">
While the sever terminal seems fine.
<img width="863" alt="p4" src="https://github.com/Abe404/root_painter/assets/150230377/57441331-0ae0-4559-9bda-6a2ccb1479fd">
Kind regards,
Zihao

(Updated on 2023.11.22)
When the process is not responding, I generated a dump file for the specific process, then I used WinDbg and chatgpt to analyize the dump file.Here's a summary of the analysis:

1. **Exception Information:**
   ```
   EXCEPTION_CODE: (NTSTATUS) 0x80000003 - {    }
   ```
   This indicates a breakpoint exception, suggesting that the program encountered a breakpoint during execution.

2. **Stack Trace:**
   ```
   STACK_TEXT:  
   000000f7`4b1e2be8 00007ff8`9d71cfda     : 00000000`00000000 00007ff8`9d7181f2 000000f7`4b1e2d20 00000294`8357cab0 : ntdll!NtOpenFile+0x14
   ...
   ```
   The stack trace shows the function call path at the time of the crash, with `ntdll!NtOpenFile` potentially being the last called function.

3. **Module Information:**
   ```
   MODULE_NAME: python38
   IMAGE_NAME:  python38.dll
   ```
   The exception occurred in the python38.dll module, which is the core library of the Python interpreter.

4. **Crash Point Information:**
   ```
   SYMBOL_NAME:  python38+180f
   ```
   The crash point is located at offset 0x180f in the python38.dll module.

Based on this information, one possible cause is that the program attempted to execute a breakpoint, which could be triggered by debugging operations or certain conditions. In normal operation, encountering a breakpoint exception is unexpected.

The complete analysis results of windbg are in the txt file below.
[Exception Analysis  .txt](https://github.com/Abe404/root_painter/files/13439365/Exception.Analysis.txt)

**Comment by Abe404** [comment]
I think we discussed this via email also? If I recall correctly, you identified the problem with the file sync and fixed it. Let me know if you still need more input.

**Reply by qielax** [reply]
Yes, we discussed the problem via email and I fixed it. Thank you for your help！
